src/main/scala/cromwell/engine/SymbolTable.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.engine
+
+import cromwell.binding.types.WdlType
+import cromwell.binding.{FullyQualifiedName, Call, WdlBinding, WdlValue}
+import scala.collection.mutable
+
+class SymbolTableEntry(val scope: String, val name: String, val iteration: Int, var wdlValue: Option[WdlValue], val wdlType: WdlType, input: Boolean) {
+  def isInput: Boolean = input
+
+  def isOutput: Boolean = !isInput
+}
+
+class SymbolTable(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]) {
+  val table = mutable.Set[SymbolTableEntry]()
+
+  binding.workflow.calls.foreach { call =>
+    call.task.inputs.foreach { case (inputName, wdlType) =>
+      table += new SymbolTableEntry(call.fullyQualifiedName, inputName, 0, None, wdlType, input = true)
+    }
+    call.task.outputs.foreach { output =>
+      table += new SymbolTableEntry(call.fullyQualifiedName, output.name, 0, None, output.wdlType, input = false)
+    }
+  }
```

Comment:
I think the above block is definitely fine for this ticket.  We should also keep in mind that `binding.workflow.calls` will become `binding.workflow.children` (where `children` is a `Set[Scope]`) at some point.